examples/guestbook-operator/Makefile

```diff
@@ -56,7 +61,9 @@ generate: controller-gen
 
 # Build the docker image
 docker-build: test
+	sed -e "s:^replace://replace:g" go.mod > go.mod.make-docker-build
```

Comment:
This is a rough approach, I think...
Without this change, `go run smoketest.go` will fail.
Here is log.
```
...
...
Step 5/14 : RUN go mod download
 ---> Running in 3e6c6d104896
go: sigs.k8s.io/kubebuilder-declarative-pattern@v0.0.0-20200317144824-bbf1fb2a4a9a: parsing /go.mod: open /go.mod: no such file or directory
The command '/bin/sh -c go mod download' returned a non-zero code: 1
Makefile:64: recipe for target 'docker-build' failed
make: *** [docker-build] Error 1
```

Any Ideas? or, is this reasonable?

Reply:
Hey @atoato88 ! I'm trying to think of a different solution here, we may not have much of an option though. It's a tricky scenario to build correctly.

Reply:
The problem is that the build target is dependent on a library that is in the same source tree (both with their own go mod files). The separation is necessary because we don't want examples/.. to be part of the library when a user consumes this.

The dashboard-operator took the following approach:

- Rename the kubebuilder-declarative-pattern in the go.mod to ../../: [src](https://github.com/kubernetes-sigs/kubebuilder-declarative-pattern/blob/bbf1fb2a4a9a5ab3cf14d3f5f3f06658ca6373f8/examples/dashboard-operator/go.mod#L13) 
- Use vendored dependencies (instead of go modules) in the docker build [src](https://github.com/kubernetes-sigs/kubebuilder-declarative-pattern/blob/bbf1fb2a4a9a5ab3cf14d3f5f3f06658ca6373f8/examples/dashboard-operator/Dockerfile#L13)
- Update the vendor dependencies as part of docker-build [src](https://github.com/kubernetes-sigs/kubebuilder-declarative-pattern/blob/bbf1fb2a4a9a5ab3cf14d3f5f3f06658ca6373f8/examples/dashboard-operator/Makefile#L56)

This approach ensures that you are always running the latest code from the main repo when running the example. This is low maintenance, great for testing/development. It would 'feel better' if it didn't require using vendored dependencies. There may be a workaround for that, but I'm not sure.

Reply:
Thank you for helpful comment, I'll change my PR according to comment!! :smile:

Reply:
I've fixed about this.
Your comment solved all concerning.
Thank you!!